lib/Doctrine/DBAL/Driver/PDOStatement.php

```diff
@@ -2,40 +2,71 @@
 
 namespace Doctrine\DBAL\Driver;
 
+use IteratorAggregate;
+use PDO;
+
 /**
  * The PDO implementation of the Statement interface.
  * Used by all PDO-based drivers.
  *
  * @since 2.0
  */
-class PDOStatement extends \PDOStatement implements Statement
+class PDOStatement implements IteratorAggregate, Statement
 {
     /**
-     * Protected constructor.
+     * @var array<int,int>
```

Comment:
This format isn't generally well accepted by SA tools, it should be `int[]`.

Reply:
I may have misinterpreted https://github.com/doctrine/dbal/pull/2495#discussion_r96125973. And FWIW, PHPStan supports this syntax:

https://github.com/phpstan/phpstan/blob/84b09eba4f2a4f22faa03fc4c94bc34c89bf02a9/tests/PHPStan/Analyser/data/foreach/foreach-with-specified-key-type.php#L9

From the domain standpoint, it's a map of integers to integers, not a list of integers. Do you still believe it should be `int[]`?

Reply:
I understand, but this syntax does not work in PhpStorm at all (recognized as mixed => mixed) or even our CS. `array<>` syntax has never been part od phpDoc.

Reply:
Agree, will rework.